group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -420,6 +423,18 @@ public Group group(String groupId, long committedOffset) throws GroupIdNotFoundE
         return group;
     }
 
+    /**
+     * @return The GenericGroup List filtered by statesFilter or typesFilter.
+     */
+    public ListGroupsResponseData listGroups(ListGroupsRequestData request, long committedOffset) {
+        Stream<Group> groupStream = groups.values(committedOffset).stream();
+        List<String> statesFilter = request.statesFilter();
+        if (!statesFilter.isEmpty()) {
+            groupStream = groupStream.filter(group -> statesFilter.contains(group.stateAsString()));
+        }
+        return new ListGroupsResponseData().setGroups(groupStream.map(Group::asListedGroup).collect(Collectors.toList()));
```

Comment:
In the ConsumerGroup case, the state is stored in a timeline data structure. Hence, we need to pass the `committedOffset` to `stateAsString` and `asListedGroup` in order to stay consistent.

Reply:
done.